blocklogger.go

```diff
@@ -97,6 +97,61 @@ func (b *blockProgressLogger) logBlockHeight(block *dcrutil.Block) {
 	b.lastBlockLogTime = now
 }
 
+// logBlockHeightForce logs a new block height as an information message to show
+// progress to the user. It is identical to the logBlockHeight above, except
+// it does NOT limit logging to one message every 10 seconds with duration and
+// totals included.
```

Comment:
It's probably better to have one function be based on the other, given this kind of copy+pasted function tends to get out of sync from each other fast without very careful review.

Reply:
I most definitely agree, but the issue is that the original function (logBlockHeight) returns prematurely if the duration condition is not satisfied. We want to force a log when the sync is complete regardless of duration passed. I've explored a bunch of alternate methods for resolving this issue, and there is the obvious one of modifying the logBlockHeight function parameters to include a reference to, or something with a reference to, the syncHeight. However, this might not be desirable.

Perhaps we can, since we are unlikely to call logBlockHeightForce for anything again, add that code directly to the blockmanager.go file. This will eradicate the need for a "copy-and-paste" function and eliminate the possibility of any "out-of-sync" functions that we fear for.

What do you think?

Reply:
> add that code directly to the blockmanager.go

Do you mean copy the contents of `logBlockHeightForce()` into the point where you added the if? That would be worse IMO.

I'd personally just do the simple thing and pass a `lastBlock bool` to `logBlockHeight()`, then _not_ return from the function if the bool is set. Then adjust call sites as needed.

Reply:
Your right, perhaps the simple way is the best way in the case. Appropriate changes have been made.